cli/azd/internal/appdetect/docker.go

```diff
@@ -1,19 +1,69 @@
 package appdetect
 
 import (
+	"bufio"
 	"io/fs"
+	"os"
 	"path/filepath"
+	"regexp"
+	"strconv"
 	"strings"
 )
 
 func detectDocker(path string, entries []fs.DirEntry) (*Docker, error) {
 	for _, entry := range entries {
 		if strings.ToLower(entry.Name()) == "dockerfile" {
+			dockerFilePath := filepath.Join(path, entry.Name())
+			exposedPorts, _ := getExposedPorts(dockerFilePath)
 			return &Docker{
-				Path: filepath.Join(path, entry.Name()),
+				Path:         dockerFilePath,
+				ExposedPorts: exposedPorts,
 			}, nil
 		}
 	}
 
 	return nil, nil
 }
+
+func getExposedPorts(dockerfilePath string) (map[int]string, error) {
+	file, err := os.Open(dockerfilePath)
+	if err != nil {
+		return nil, err
+	}
+	defer file.Close()
+
+	ports := make(map[int]string)
+	scanner := bufio.NewScanner(file)
+	exposeRegex := regexp.MustCompile(`^EXPOSE\s+(.+)$`)
```

Comment:
Let's avoid using regex here if we can. Some ideas from the other [comment](https://github.com/Azure/azure-dev/pull/4454/files#r1806657892).

Reply:
Updated.